mixin/smarty-v2@1/mixin.php

```diff
@@ -26,7 +26,7 @@
     $smarty = CRM_Core_Smarty::singleton();
     $v2 = isset($smarty->_version) && version_compare($smarty->_version, 3, '<');
     $templateDirs = (array) ($v2 ? $smarty->template_dir : $smarty->getTemplateDir());
-    $templateDirs = array_merge($newDirs, $templateDirs);
+    $templateDirs = array_unique(array_merge($newDirs, $templateDirs));
```

Comment:
Similar to the other PR, this won't uniqify on windows, e.g. I have two entries for the same template dir that look like

`...sites/default/files/civicrm\ext/thing/templates`
`...sites\default\files\civicrm\ext\thing\templates\`

And besides the slash direction, note the second has a trailing and the other doesn't.

Reply:
something like

```php
$templateDirs = array_merge($newDirs, $templateDirs);
$templateDirs = array_unique(array_map(function($v) {
  $v = str_replace(DIRECTORY_SEPARATOR, '/', $v);
  $v = rtrim($v, '/') . '/';
  return $v;
}, $templateDirs));
```

Reply:
Seems reasonable to me. Haven't tested on Windows, but it still works on my Linux. Added that on top.

Reply:
Thanks. I suppose could micro-optimize it further if it still seems slow - e.g. it doesn't need to re-normalize the paths that are already in templateDirs that it normalized before.